src/app/backups/views/BackupsView/BackupsView.tsx

```diff
@@ -1,90 +1,89 @@
   const devices = useAppSelector((state) => state.backups.devices);
   const currentDevice = useAppSelector((state) => state.backups.currentDevice);
+  const [foldersInBreadcrumbs, setFoldersInBreadcrumbs] = useState<DriveFolderData[]>([]);
 
-  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
-  const [selectedDevices, setSelectedDevices] = useState<(Device | DriveFolderData)[]>([]);
-  const [backupsAsFoldersPath, setBackupsAsFoldersPath] = useState<DriveFolderData[]>([]);
+  const {
+    folderUuid,
+    isFileViewerOpen,
+    itemToPreview,
+    selectedItems,
+    onFolderUuidChanges,
+    clearSelectedItems,
+    onCloseFileViewer,
+    onItemSelected,
+    onItemClicked,
+    onSelectedItemsChanged,
+  } = useBackupListActions(setFoldersInBreadcrumbs, dispatch);
 
-  const onDeviceClicked = (target: Device | DriveFolderData) => {
-    setSelectedDevices([]);
-    dispatch(backupsActions.setCurrentDevice(target));
-    if ('mac' in target) {
-      dispatch(backupsThunks.fetchDeviceBackupsThunk(target.mac));
-    }
-  };
+  const {
+    selectedDevices,
+    isDeleteModalOpen,
+    goToFolder,
+    goToFolderRoot,
+    onConfirmDelete,
+    onDeviceClicked,
+    onDevicesSelected,
+    onOpenDeleteModal,
+    onCloseDeleteModal,
+  } = useBackupDeviceActions(onFolderUuidChanges, setFoldersInBreadcrumbs, dispatch);
 
-  const onOpenDeleteModal = (targets: (Device | DriveFolderData)[]) => {
-    setSelectedDevices((values) => [...values, ...targets]);
-    setIsDeleteModalOpen(true);
-  };
+  const { currentItems, areFetchingItems, hasMoreItems, getMorePaginatedItems, getFolderContent } =
+    useBackupsPagination(folderUuid, clearSelectedItems);
 
-  const onDevicesSelected = (changes: { device: Device | DriveFolderData; isSelected: boolean }[]) => {
-    let updatedSelectedItems = selectedDevices;
-    for (const change of changes) {
-      updatedSelectedItems = updatedSelectedItems.filter((item) => item.id !== change.device.id);
-      if (change.isSelected) {
-        updatedSelectedItems = [...updatedSelectedItems, change.device];
-      }
-    }
-    setSelectedDevices(updatedSelectedItems);
+  const onDownloadSelectedItems = () => {
+    dispatch(downloadItemsThunk(selectedItems));
   };
 
-  const onConfirmDelete = async () => {
-    for (const selectedDevice of selectedDevices) {
-      if (selectedDevice && 'mac' in selectedDevice) {
-        dispatch(backupsThunks.deleteDeviceThunk(selectedDevice));
+  async function onDeleteSelectedItems() {
+    for (const item of selectedItems) {
+      if (item.isFolder) {
+        await deleteBackupDeviceAsFolder(item as DriveWebFolderData);
       } else {
-        await dispatch(deleteItemsThunk([selectedDevice as DriveItemData])).unwrap();
-        await deleteBackupDeviceAsFolder(selectedDevice as DriveWebFolderData);
-        dispatch(backupsThunks.fetchDevicesThunk());
+        await deleteFile(item);
```

Comment:
To use a concurrent deletion maybe could do it like this:

 ```
 const deletePromises = selectedItems.map(item => 
      item.isFolder
        ? deleteBackupDeviceAsFolder(item as DriveWebFolderData)
        : deleteFile(item)
    );
    await Promise.all(deletePromises);
```
This will be faster than the sequential solution, especially for large deletions